pkg/database/auditable.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package database
+
+// Auditable represents a resource that can be audited as an actor or actee.
+type Auditable interface {
```

Comment:
My knowledge of gorm is limited. Is there anyway to do a callback here so that we can increment a counter on every audit entry inserted? If there is, can we do it?

Reply:
We have access to the tx in the callback, so we can increment a SQL counter (that's how we do the in-app stats actually). 

We can create a global callback that has access to metrics and then basically guard it with a `if table == "audit_entries"`. What do you hope to glean from counting the number of audits?

Reply:
I would love a graph in our dashboards to see how many things are happening on a server from an admin point of view. Basically a stackdriver stat that says x number of actions lets us see the rate of actions happening on an instance. Good for debugging or tracking admin abuse.